Assets/MixedRealityToolkit.SDK/Experimental/Dialog/Scripts/Dialog.cs

```diff
@@ -0,0 +1,232 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Dialog
+{
+    public abstract class Dialog : MonoBehaviour
+    {
+        protected DialogResult result;
+
+        protected DialogState state = DialogState.Uninitialized;
+        public DialogState State
+        {
+            get
+            {
+                return state;
+            }
+            set
+            {
+                state = value;
+            }
+        }
+
+        /// <summary>
+        /// Called after user has clicked a button and the dialog has finished closing
+        /// </summary>
+        public Action<DialogResult> OnClosed;
+
+        /// <summary>
+        /// Can be used to monitor result instead of events
+        /// </summary>
+        public DialogResult Result
+        {
+            get
+            {
+                return result;
+            }
+        }
+
+        protected void Launch(DialogResult newResult)
+        {
+            if (State != DialogState.Uninitialized)
+            {
+                return;
+            }
+
+            result = newResult;
+            StartCoroutine(RunDialogOverTime());
+        }
+
+        /// <summary>
+        /// Opens dialog, waits for input, then closes
+        /// </summary>
+        /// <returns></returns>
+        protected IEnumerator RunDialogOverTime()
+        {
+            // Create our buttons and set up our message
+            GenerateButtons();
+            SetTitleAndMessage();
+            FinalizeLayout();
+
+            // Open dialog
+            State = DialogState.Opening;
+            yield return StartCoroutine(OpenDialog());
+            State = DialogState.WaitingForInput;
+            // Wait for input
+            while (State == DialogState.WaitingForInput)
+            {
+                UpdateDialog();
+                yield return null;
+            }
+            // Close dialog
+            State = DialogState.Closing;
+            yield return StartCoroutine(CloseDialog());
+            State = DialogState.Closed;
+            // Callback
+            if (OnClosed != null)
+            {
+                OnClosed(result);
+            }
+            // Wait a moment to give scripts a chance to respond
+            yield return null;
+            // Destroy ourselves
+            GameObject.Destroy(gameObject);
+            yield break;
+        }
+
+        /// <summary>
+        /// Opens the dialog - state must be set to WaitingForInput afterwards
+        /// Overridden in inherited class.
+        /// </summary>
+        /// <returns></returns>
+        protected virtual IEnumerator OpenDialog()
+        {
+            yield break;
```

Comment:
Confused what these are for? or jsut to-be-filled?

Reply:
It's just to-be-filled.
We decided to keep the logic of the previous version of Mixed Reality Toolkit so to simplify the porting of existing applications